lib/Sema/CSStep.cpp

```diff
@@ -356,34 +356,36 @@ StepResult ComponentStep::take(bool prevFailed) {
   });
 
   auto *disjunction = CS.selectDisjunction();
-  auto *conjunction = CS.selectConjunction();
 
   if (CS.isDebugMode()) {
-    PrintOptions PO;
-    PO.PrintTypesForDebugging = true;
-
-    auto &log = getDebugLogger();
     if (!potentialBindings.empty()) {
+      auto &log = getDebugLogger();
       log << "(Potential Binding(s): " << '\n';
       log << potentialBindings;
     }
-    log.indent(CS.solverState->getCurrentIndent());
-
-    if (disjunction) {
+    
+    SmallVector<Constraint *, 4> disjunctions;
+    CS.collectDisjunctions(disjunctions);
+    if (!disjunctions.empty()) {
+      PrintOptions PO;
+      PO.PrintTypesForDebugging = true;
+      
+      auto &log = getDebugLogger();
       log.indent(2);
       log << "Disjunction(s) = [";
-      auto constraints = disjunction->getNestedConstraints();
-      log << constraints[0]->getFirstType()->getString(PO);
-      log << "]";
+      std::vector<std::string> types;
+      for (const auto &disjunction : disjunctions) {
+        auto constraints = disjunction->getNestedConstraints();
+        types.push_back(constraints[0]->getFirstType()->getString(PO));
```

Comment:
We need to do some more checking here. I'd say the most interesting disjunctions are the one with `BindOverload` choices, let's filter the list based on that, it would also make sure that first type exists and it's always a type variable.